device/ospf/ospfv2/interface/interface.yaml

```diff
@@ -0,0 +1,225 @@
+components:
+  schemas:
+    Ospfv2.Interface:
+      description: >-
+        Configuration for single OSPFv2 interface.
+      type: object
+      required: [ipv4_name, name]
+      properties:
+        name:
+          x-include: ../common/common.yaml#/components/schemas/Named.Object/properties/name
+          x-field-uid: 1
+        ipv4_name:
+          description: >-
+            The globally unique name of the IPv4 interface connected to the DUT.
+            This name must match the "name" field of the "ipv4_addresses" on top which this OSPFv2 interface is configured.          
+          type: string
+          x-constraint:
+          - '/components/schemas/Device.Ipv4/properties/name'
+          x-field-uid: 2
+        choice:
```

Comment:
I think choice is more like a keyword. Might be better to have intf.area.number = 0 ( default) / optional intf.area.ip = 0.0.0.0 (non-default)